src/System.Private.ServiceModel/tools/scripts/StopWcfSelfHostedSvc.cmd

```diff
@@ -25,10 +38,38 @@ if '%_runelevated%' == '' (
 )
 
 if NOT [%ERRORLEVEL%]==[0] (
-	echo Warning: An error occurred while killing the SelfHostedWCFService.exe. >>%_cleanuplog%
-	exit /b 1
+  _EXITCODE=%ERRORLEVEL%
+	echo Warning: An error occurred attempting to stop the SelfHostedWCFService.exe. >>%_cleanuplog%
+	goto done
 	)
 	
-exit /b 0
+REM Terminating the Self host WCF process started with StartSelfHostWcfSvc.cmd
+REM results in that script gaining control to finish cleanup.  If we are in that
+REM state, we need to wait for that cleanup to finish, or the bin folder will still
+REM be in use.
+
+for /l %%x in (1, 1, %_maxWaitIterations%) do (
+  echo Waiting for Self hosted WCF service cleanup to complete... >>%_cleanuplog%
+  if NOT EXIST %_setupMonitorFile% (
+    echo Self hosted WCF service cleanup has completed. >>%_cleanuplog%
+    goto done
+  )
+  
+  REM The "proper" way to wait is 'timeout' but that unconditionally
+  REM exits the process with "Input redirection is not supported".
+  REM The community preferred approach to wait is this.
+  ping 127.0.0.1 -n 6 > nul
+
+echo %%x
+
+)
+
+REM Timed out waiting for cleanup script to finish
+echo Failed to finish cleanup for Self hosted WCF service
+set _EXITCODE=1
+
+:done
 
+if EXIST %_setupMonitorFile% del %_setupMonitorFile%
```

Comment:
If this file still exists at this point, it means something went wrong. This file is not created by this script. Should this script delete it?